src/app/api/analytics/storage.py

```diff
@@ -29,14 +27,48 @@ class SearchStore:
     missing_materials: dict[uuid.UUID, CountStatistics]
 
 
+@dataclass(frozen=True)
```

Comment:
these are more or less dupliates of the BaseModels in `analytics.py`? Any chance to deduplicate this? Alternatively document why we need to duplicate them as dataclasses?

Reply:
One is a dataclass one is a pydantic model. Either I construct a way of mapping one to the other or I drop the data class in favour of the pydantic model. Any preferences?